src/Workspaces/MSBuild/Core/MSBuild/MSBuildProjectLoader.SolutionReader.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text.Json;
+using Roslyn.Utilities;
+using Microsoft.VisualStudio.SolutionPersistence;
+using Microsoft.VisualStudio.SolutionPersistence.Model;
+using Microsoft.VisualStudio.SolutionPersistence.Serializer;
+using System.Threading;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.CodeAnalysis.MSBuild
+{
+    public partial class MSBuildProjectLoader
+    {
+        private static class SolutionReader
+        {
+            public static bool IsSlnFilename(string filename)
+            {
+                ISolutionSerializer ? serializer = SolutionSerializers.GetSerializerByMoniker(filename);
+                return serializer == null ? false : true;
+            }
+            //Rename the filter files and user proper namming convension
```

Comment:
nit: newline following closing brace
```suggestion

            //Rename the filter files and user proper naming convention
```